build.rs

```diff
@@ -383,6 +401,19 @@ fn build() -> io::Result<()> {
     Ok(())
 }
 
+fn os_from_triple(triple: &str) -> &str {
```

Comment:
Is there really no way to use rust's `target_os!` or something more standard? I'm afraid this will bite us if other architectures start to pop up

Reply:
Rust's target definitions are Rust's, and have no guarantee of compatibility with anything, e.g. Rust uses `macos`, but the rest of the world uses `darwin`.

Reply:
would it be possible to have something that just checks if `target_os!` is `macos`, and replace it by `darwin` if it is? And maybe that does that for other OSes, if it happens for other?

Reply:
I suspect it's needed for more OSes, but I don't have them to test. Whatever you do, it's going to be an incomplete list that may break on OSes that haven't been tested.

Reply:
Quick question, but what's the value of `CARGO_CFG_TARGET_OS` on macos? Can't you just use a wrapper that gets CARGO_CFG_TARGET_OS and if it's macos, replace it by darwin?

Reply:
`CARGO_CFG*` are derived from rustc's cfg values, so they use the same naming scheme.

I can hardcode the result to be correct only on darwin/macos, and leave it incorrect everywhere else, but why should I leave it incorrect everywhere else?

Reply:
my question is just, why do this triple splitting magic (and explode if it doesn't work) instead of just using `CARGO_CFG_TARGET` and use a hashmap or something like that to substitute the wrong stuff with whatever you know is right? Or maybe I'm missing the point here?

Reply:
Split doesn't work, because Rust's triples are inconsistently named, and sometimes have the "vendor" and sometimes they don't, so you don't know where the OS part starts.

HashMap would work for a closed set. I'm trying to make it work for an open set.

Reply:
my point was that maybe we could do a hashmap like `{'macos': 'darwin'}` and then just have a custom `target_os` function that would just return either the string itself, if it's not in the hashmap? Like
```
fn target_os(target: &str) -> &str {
    let mut os_replacements = HashMap::new();

    os_replacements.insert(
        "macos".to_string(),
         "darwin".to_string(),
     );
     os_replacements.get(&target).unwrap_or("target")
}
```
But I feel like we're talking about different things though, so I could be completely wrong :smile: 

https://github.com/rustsec/rustsec/blob/main/platforms/src/target/os.rs#L112-L134 seems to have a pretty consistent list of OS if we wanna list more stuff here?